redbot/cogs/warnings/warnings.py

```diff
@@ -386,6 +387,34 @@ async def warn(
         or a custom reason if ``[p]warningset allowcustomreasons`` is set.
         """
         guild = ctx.guild
+        member = None
+        if isinstance(user, discord.Member):
+            member = user
+        elif isinstance(user, int):
+            user_obj = self.bot.get_user(user) or discord.Object(id=user)
```

Comment:
Rather than asking and hoping we have perms, let's add a check to exit slightly earlier if the bot doesn't have them
```suggestion
            if not ctx.channel.permissions_for(ctx.guild.me).ban_members:
                await ctx.send(_("User {user} is not in the server."),format(user=user))
                return
            user_obj = self.bot.get_user(user) or discord.Object(id=user)
```